### PR TITLE
simple substep implemenation

### PR DESCRIPTION
**Pull request by misterbrownlee**
Greetings impress.js overlords:

I have made an unobtrusive and clean attempt at providing the basics of a substep feature.  

My implementation follows the syntax and behavior modeled by regular steps:
- denote a substep by adding the css class 'substep' to elements in a step
- init will prepare the substeps with the additional css class 'future'
- next/prev will check within the current step for potential substeps
- navigation through substeps will set 'present', 'active', and 'past' - just like steps
- the only variation is substeps remain 'active' when past
- backward navigation removes 'past/active', restores 'future' as needed
- I've included substep events, following a similar model as regular steps

Because the only presentation-side change is CSS classes, there would be no effect on existing presentations.  A new presentation would have to implement substep specific css to utilize the feature.  This is outlined in the comments of the example html and css. 

I thought it was best to create a separate example, rather than tinker with the existing index.html, as it seems like someone who wants substeps will find the html and figure it out from there.

Substeps are not included as a separate API feature.  I felt this was the right thing because it didn't seem necessary to directly navigate to a substep. I also didn't think deep linking a substep made sense, and the sytax for that would have gotten weird, anyway.  It did make sense to hook into next/prev. If you're adding substeps, you probably wouldn't want prev/next to skip them, even if you're moving via the API.  

I felt an expected default behavior was to have substeps remain active, but that's easily changed.  This could also be easily toggled with a 'data-' property.

The TL;DR notes on my commit are:
- a couple of helper functions for finding and handling substeps
- a small if/else check in next/prev to decide if a substep is available
- a separate example html page with it's own CSS. 

I know substeps are 'powerpointish', and thus subject to controversy.  They also may not meet the impress.js philosophy.  My coding style might not be exactly in-line with the main repo.  If the philosophical issue can be overcome, I'm happy to revise my code as needed to be acceptable.  Just let me know.

Thanks for considering my contribution! 

-- Aaron (aka tehfoo)

### CODE REVIEW

**Comment by FagnerMartinsBrack** [comment]
Hi @misterbrownlee , in an effort to clear up older issues/PRs we are pinging back to know if you are still tracking this request.

To give a little bit of context, recently [a decision was made in the project](https://github.com/impress/impress.js/issues/435#issuecomment-177602861) to make the development more active and the first task is to clear up older PRs like this one to see if [the OP](http://netforbeginners.about.com/od/internetlanguage/f/What-Is-OP.htm) is still interested in keep it going.

By the way, this one seems duplicate of #214 and #264.

**Comment by carljm** [comment]
This looks like pretty much the same thing as #264, just with slightly different naming choices. I think it's a good approach in general, and I don't care which one is merged.

**Comment by FagnerMartinsBrack** [comment]
Closing due to the lack of feedback from the OP. If you want to work in the PR again, feel free to comment here to open up the discussion again.

Also, this PR was opened in the dev branch, as from the CONTRIBUTING.md file, it should be opened against the [master branch instead](https://github.com/impress/impress.js/blob/d13428ec226bcb5f6cb918b854507fde1856d042/CONTRIBUTING.md#pull-requests).
